### PR TITLE
Release changelog for 2.42

### PR DESCRIPTION
**Pull request by sergiusens**
Signed-off-by: Sergio Schvezov <sergio.schvezov@canonical.com>

- [x] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] If this is a bugfix. Have you checked that there is a bug report open for the issue you are trying to fix on [bug reports](https://bugs.launchpad.net/snapcraft)?
- [x] If this is a new feature. Have you discussed the design on the [forum](https://forum.snapcraft.io)?
- [x] Have you successfully run `./runtests.sh static`?
- [x] Have you successfully run `./runtests.sh unit`?

-----

### CODE REVIEW

File: tests/unit/project_loader/grammar_processing/test_part_grammar_processor.py [hunk 1]
```diff
@@ -226,6 +227,15 @@ class PartGrammarBuildSnapsTestCase(unit.TestCase):
 
     scenarios = multiply_scenarios(source_scenarios, arch_scenarios)
 
+    def setUp(self):
+        super().setUp()
+
+        fake_snapd = FakeSnapd()
+        fake_snapd.find_result = [{
+            'hello': {'channels': {
+                'latest/stable': {'confinement': 'devmode'}}}}]
```
Comment (kyrofa) [inline tests/unit/project_loader/grammar_processing/test_part_grammar_processor.py]: Eh? Why is this needed?

File: .travis.yml [hunk 2]
```diff
@@ -24,6 +24,8 @@ jobs:
       script: sudo ./tools/travis/run_tests.sh tests/unit
     - if: type != cron
       script: SNAPCRAFT_TEST_MOCK_MACHINE=armv7l sudo ./tools/travis/run_tests.sh tests/unit
+    - if: type != cron
+      script: SNAPCRAFT_TEST_MOCK_MACHINE=aarch64 sudo ./tools/travis/run_tests.sh tests/unit
```
[the remark this answers is not in the document]
Comment (kalikiana) [reply .travis.yml]: I was going by the architectures we test, which now includes arm64. The `ValidArchitecturesYamlTestCase` would've failed early with this place. But we don't test on ppc.
